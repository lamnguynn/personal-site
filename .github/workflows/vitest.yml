name: Run unit and UI tests on project

on:
    push:
      branches:
        - "main"
    pull_request:

jobs:
  run-vitest:
    name: Run vitest
    runs-on: ubuntu-latest
    steps:
      - name:  Checkout code
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
      - name: Setup Node
        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
        with:
          node-version: 24
      - name: Install dependencies
        run: |
            npm ci
      - name: Run unit tests
        run: |
          npm run test-unit
      - name: Run UI tests
        run: |
          npm run test-ui