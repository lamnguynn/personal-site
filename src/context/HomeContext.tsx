import {
  useState,
  createContext,
  type Dispatch,
  type SetStateAction,
} from 'react';
import type { Vector3 } from '@/types/RaceCircuitType';

type IsCarAtTarget = boolean;
type SectionTitle = string | undefined;
type Index = number | undefined;
type CarTarget = {
  label: string | undefined;
  coordinate: Vector3 | undefined;
};

export const HomeContext = createContext<
  | {
      isCarAtTarget: IsCarAtTarget;
      setIsCarAtTarget: Dispatch<SetStateAction<IsCarAtTarget>>;
      sectionTitle: SectionTitle;
      setSectionTitle: Dispatch<SetStateAction<SectionTitle>>;
      animateRowIndex: Index;
      setAnimateRowIndex: Dispatch<SetStateAction<Index>>;
      carTarget: CarTarget;
      setCarTarget: Dispatch<SetStateAction<CarTarget>>;
    }
  | undefined
>(undefined);

export function HomeProvider({ children }: { children: React.ReactNode }) {
  const [isCarAtTarget, setIsCarAtTarget] = useState<IsCarAtTarget>(false);
  const [sectionTitle, setSectionTitle] = useState<SectionTitle>(undefined);
  const [animateRowIndex, setAnimateRowIndex] = useState<Index>(undefined);
  const [carTarget, setCarTarget] = useState<CarTarget>({
    label: undefined,
    coordinate: undefined,
  });

  return (
    <HomeContext.Provider
      value={{
        isCarAtTarget,
        setIsCarAtTarget,
        sectionTitle,
        setSectionTitle,
        animateRowIndex,
        setAnimateRowIndex,
        carTarget,
        setCarTarget,
      }}
    >
      {children}
    </HomeContext.Provider>
  );
}
