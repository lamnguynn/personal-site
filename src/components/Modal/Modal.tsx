import { HomeContext } from '@/context/HomeContext';
import { getContent } from '@/util/util';
import { useContext } from 'react';
import Content from '@/components/Content/Content';

export default function Modal({}) {
  const context = useContext(HomeContext);
  const { animateRowIndex, sectionTitle } = context!;

  return (
    <>
      {animateRowIndex !== undefined && (
        <div className="absolute bottom-0 left-[10%] right-[10%] md:left-[20%] md:right-[20%] lg:left-[30%] lg:right-[30%] h-auto bg-amber-200 z-[100000000] p-4 pb-0">
          <div className="relative bg-amber-300 w-full h-full overflow-scroll">
            <Content
              sectionTitle={sectionTitle}
              data={getContent(animateRowIndex).data}
            />
          </div>
        </div>
      )}
    </>
  );
}
