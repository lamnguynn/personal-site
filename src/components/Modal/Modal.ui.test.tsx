import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { HomeProvider } from '@/context/HomeContext';

import Modal from './Modal';

describe('Modal component', () => {
  test('snapshot', () => {
    const { asFragment } = render(
      <HomeProvider>
        <Modal />
      </HomeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
