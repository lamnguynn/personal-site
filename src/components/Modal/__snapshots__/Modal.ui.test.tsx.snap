// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal component > snapshot 1`] = `<DocumentFragment />`;
